src/imitation/algorithms/bc.py

```diff
@@ -132,63 +197,32 @@ def train(
         """
         assert self.batch_size >= 1
         samples_so_far = 0
+        batch_num = 0
         for epoch_num in trange(n_epochs, desc="BC epoch"):
             while samples_so_far < (epoch_num + 1) * self.expert_dataset.size():
```

Comment:
This is okay, but in future it would be nice to create a common batch iterator wrapper for datasets so that we can simplify this sort of code.

(hold off on this change; if we move to PyTorch then we can probably reuse their dataset abstractions)